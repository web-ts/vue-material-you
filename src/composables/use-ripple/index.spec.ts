import { expect, describe, it, vi } from "vitest";

import useRipple from ".";

describe("use-ripple", () => {
  const touchEvent = {
    target: {
      getBoundingClientRect() {
        return { left: 0, top: 0, width: 100, height: 50 };
      }
    },
    touches: [
      {
        clientX: 100,
        clientY: 100
      }
    ]
  };

  it("should match default values", () => {
    const { rippleStyleObject } = useRipple();

    expect(rippleStyleObject.value).toMatchSnapshot();
  });

  it("should expand on click and match the client events", async () => {
    const { rippleStyleObject, events } = useRipple();

    events.onTouchdown(touchEvent as any);

    expect(rippleStyleObject.value).toMatchSnapshot();
  });

  it("should revet to default button size", async () => {
    vi.spyOn(global, "setTimeout").mockImplementation((fn) => fn() as any);
    const { rippleStyleObject, events } = useRipple();

    events.onTouchdown(touchEvent as any);

    events.onTouchup(touchEvent as any);

    expect(rippleStyleObject.value).toMatchSnapshot();
  });
});
