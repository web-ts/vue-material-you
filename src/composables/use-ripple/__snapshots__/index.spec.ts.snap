// Vitest Snapshot v1

exports[`use-ripple > should expand on click and match the client events 1`] = `
{
  "--ripple-height": "50px",
  "--ripple-opacity": "0.2",
  "--ripple-pos-x": "100px",
  "--ripple-pos-y": "100px",
  "--ripple-scale": "4.4",
  "--ripple-width": "50px",
  "--transition-property": "transform",
}
`;

exports[`use-ripple > should match default values 1`] = `
{
  "--ripple-height": "40px",
  "--ripple-opacity": "0",
  "--ripple-pos-x": "0px",
  "--ripple-pos-y": "0px",
  "--ripple-scale": "1",
  "--ripple-width": "40px",
  "--transition-property": "transform",
}
`;

exports[`use-ripple > should revet to default button size 1`] = `
{
  "--ripple-height": "50px",
  "--ripple-opacity": "0",
  "--ripple-pos-x": "0px",
  "--ripple-pos-y": "0px",
  "--ripple-scale": "1",
  "--ripple-width": "50px",
  "--transition-property": "opacity",
}
`;
